src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts

```diff
@@ -664,5 +653,6 @@ export class LocalProcessExtensionHost implements IExtensionHost {
 			this._extensionHostDebugService.terminateSession(this._environmentService.debugExtensionHost.debugId);
 			event.join(timeout(100 /* wait a bit for IPC to get delivered */), 'join.extensionDevelopment');
 		}
+		this.terminate();
```

Comment:
Should this `terminate` call `join` the `onWillShutdown` to ensure the message is delivered before the app goes down? I think `terminate` itself runs a `promise.then` but does not signal that to the outside:

https://github.com/microsoft/vscode/blob/f250472adc449427b8b06ea77f23d780c982a40a/src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts#L624-L624

It also seems to me that `terminate` is called from more locations:
* https://github.com/microsoft/vscode/blob/f250472adc449427b8b06ea77f23d780c982a40a/src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts#L149-L151
* https://github.com/microsoft/vscode/blob/f250472adc449427b8b06ea77f23d780c982a40a/src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts#L142-L142

Reply:
Locally, I created a new `terminate` function that returns a promise, and then joined it in `onWillShutdown`. I noticed that the reloading took noticeably longer, but I also managed to make it so that the extension host process isn't leaked. However, there's still some hangs that occur, so I need to look more into what's happening there.